src/Blankslate/Blankslate.tsx

```diff
@@ -1,71 +1,148 @@
+  }
 
-  &[data-border='true'] {
-    border: ${get('borderWidths.1')} solid ${get('colors.border.default')};
-    border-radius: ${get('radii.2')};
+  .Blankslate[data-border='true'] {
+    border: var(--borderWidth-thin) solid var(--borderColor-default, ${get('colors.border.default')});
+    border-radius: var(--borderRadius-medium);
   }
 
-  &[data-narrow='true'] {
+  .Blankslate[data-narrow='true'] {
     margin: 0 auto;
     max-width: 485px;
   }
 
-  &[data-spacious='true'] {
-    padding: ${get('space.9')} ${get('space.6')};
+  .Blankslate-Heading,
+  .Blankslate-Description {
+    margin: 0;
+    margin-bottom: var(--stack-gap-condensed);
   }
 
-  .BlankSlateAction {
-    margin-top: ${get('space.3')};
+  .Blankslate-Heading {
+    font-size: var(--text-title-size-medium);
+    font-weight: var(--text-title-weight-medium);
   }
 
-  .BlankSlateAction:first-of-type {
-    margin-top: ${get('space.4')};
+  .Blankslate-Description {
+    color: var(--fgColor-muted, ${get('colors.fg.muted')});
+    font-size: var(--text-body-size-large);
   }
 
-  .BlankSlateAction:last-of-type {
-    margin-bottom: ${get('space.2')};
+  .Blankslate-Action {
+    margin-top: var(--stack-gap-normal);
+  }
+
+  .Blankslate-Action:first-of-type {
+    margin-top: var(--stack-gap-spacious);
+  }
+
+  .Blankslate-Action:last-of-type {
+    margin-bottom: var(--stack-gap-condensed);
   }
 `
 
-export type BlankslateProps = React.PropsWithChildren<{
-  /**
-   * Add a border around this component
-   */
-  border?: boolean
+const BlankslateContainerQuery = `
+  /* At the time these styles were written,
+  34rem was our "small" breakpoint width */
+  @container (max-width: 34rem) {
+    ${StyledBlankslate} .Blankslate {
+    --blankslate-outer-padding-block: var(--base-size-20);
+    --blankslate-outer-padding-inline: var(--base-size-20);
+  }
 
-  /**
-   * Constrain the maximum width of this component
-   */
-  narrow?: boolean
+  ${StyledBlankslate} .Blankslate[data-spacious='true'] {
+    --blankslate-outer-padding-block: var(--base-size-44);
+    --blankslate-outer-padding-inline: var(--base-size-28);
+  }
 
-  /**
-   * Increase the padding of this component
-   */
-  spacious?: boolean
-}>
+  ${StyledBlankslate} .Blankslate-Visual {
+    margin-bottom: var(--stack-gap-condensed);
+    max-width: var(--base-size-24);
+  }
+
+  ${StyledBlankslate} .Blankslate-Visual svg {
+    width: 100%;
+  }
+
+  ${StyledBlankslate} .Blankslate-Heading {
+    font-size: var(--text-title-size-small);
+  }
+
+  ${StyledBlankslate} .Blankslate-Description {
+    font-size: var(--text-body-size-medium);
+  }
+
+  ${StyledBlankslate} .Blankslate-Action {
+    margin-top: var(--stack-gap-condensed);
+  }
+
+  ${StyledBlankslate} .Blankslate-Action:first-of-type {
+    margin-top: var(--stack-gap-normal);
+  }
+
+  ${StyledBlankslate} .Blankslate-Action:last-of-type {
+    margin-bottom: calc(var(--stack-gap-condensed) / 2);
+  }
+  }
```

Comment:
Tested this PR with dotcom and other than this invalid brace, everything else passes CI!